test/connectors/httpParsers/VatObligationsHttpParserSpec.scala

```diff
@@ -80,6 +79,114 @@ class VatObligationsHttpParserSpec extends SpecBase {
       }
     }
 
+    "the http response status is 200 OK and matches expected Schema if no data" should {
+
+      val testObligations: VatObligations =
+        VatObligations(
+          Seq.empty[VatObligation]
+          )
+
+
+      val responseJson: JsValue = Json.parse(
+        """{
+          |    "obligations": []
+          |  }""".stripMargin)
+
+      val httpResponse: AnyRef with HttpResponse = HttpResponse(Status.OK, responseJson = Some(
+        responseJson
+      ))
+
+      val expected: Either[Nothing, VatObligations] = Right(testObligations)
+      val result: VatObligationsHttpParser.HttpGetResult[VatObligations] = VatObligationsReads.read("", "", httpResponse)
+
+      "return a VatObligations instance" in {
+        result shouldEqual expected
+      }
+    }
+
+
+    "response status is 200 OK and matches expected Schema when the response contains multiple obligations with multiple details" should {
```

Comment:
Is this a valid test? When would the API send back multiple `obligationDetails`?

Reply:
Absolutely, Why not?
It's an array.
The schema says you can so it is a valid test yes.
Could have more than one.

If we do the code handles it properly and get all the details.
I there are multiple obligations it pulls them out into the